backends/bmv2/expression.cpp

```diff
@@ -450,7 +450,11 @@ void ExpressionConverter::postorder(const IR::IntMod* expression)  {
 }
 
 void ExpressionConverter::postorder(const IR::Operation_Binary* expression)  {
-    binary(expression);
+    cstring op = expression->getStringOp();
+    if (op == "|+|" || op == "|-|")
+        saturated_binary(expression);
```

Comment:
Rather than doing string comparisons, you should just overload the `postorder` for AddSat and SubSat:

    void ExpressionConverter::postorder(const IR::AddSat *exp) { saturated_binary(exp); }
    void ExpressionConverter::postorder(const IR::SubSat *exp) { saturated_binary(exp); }